src/olympia/scanners/models.py

```diff
@@ -120,6 +125,32 @@ class Meta:
     def __str__(self):
         return self.name
 
+    def clean(self):
+        if self.scanner == YARA:
+            self.clean_yara()
+
+    def clean_yara(self):
+        if not self.definition:
+            raise ValidationError(_('Yara rules should have a definition'))
+
+        if 'rule {}'.format(self.name) not in self.definition:
+            raise ValidationError(
+                _(
+                    'The name of the rule in the definition should match the '
+                    'name of the scanner rule'
+                )
+            )
+
+        if len(re.findall(r'rule\s+.+?\s+{', self.definition)) > 1:
+            raise ValidationError(
+                _('Only one Yara rule is allowed in the definition')
+            )
+
+        try:
+            yara.compile(source=self.definition)
+        except Exception:
+            raise ValidationError(_('The definition is not valid'))
```

Comment:
We should probably try to re-use the exception message here if it's a `yara.SyntaxError`. Showing what the error is is going to be much more user-friendly that just a generic "not valid!" error.

Reply:
That's true. I did not find the error messages very user friendly but maybe better than nothing.

Reply:
Yeah I didn't find them super useful either but it's better than nothing, I guess.